docs/guide/dev/integration/react/README.md

```diff
@@ -215,6 +169,186 @@ component.editor.getData();
+
+Defaults to `{}`.
+
+### `data`
+
+`String`
+
+Data to be used to initially fill editor content. It's passed to {@linkapi CKEDITOR.editor#setData} method with no additional arguments.
+
+For example usage, look at [basic usage](#basic-usage) paragraph.
+
+Defaults to `''`.
+
+### `name`
+
+`String`
+
+If provided - became editor HTML element `id` and `name` properties.
+
+```jsx
+<CKEditor
+	name="myeditor"
+/>
+```
+
+Then in another place, it's possible to refer editor instance as:
+
+```js
+const editor = CKEDITOR.instances.myeditor;
+```
+
+Defaults to `undefined`.
+
+### `readOnly`
+
+`Boolean`
+
+Set the {@link features/readonly/README read-only mode}:
+
+```jsx
+<CKEditor
+	data="<p>Editor's content</p>"
+	readOnly={true}
+/>
+```
+
+This property takes precedence over the {@linkapi CKEDITOR.config#readOnly `config.readOnly`} setting.
+
+Defaults to `false`.
+
+### `style`
+
+`Object`
+
+Style rules set that will be applied to {@linkapi CKEDITOR.editor#container editor container} with {@linkapi CKEDITOR.dom.element#setStyles}.
+
+```jsx
+<CKEditor
+	data="<p>Editor's content</p>"
+	style={{
+		'margin-top': '100px',
+		'border': '5px solid red'
+	}}
+/>
+```
+
+Require [type](#type) to be `classic`.
+
+Defaults to `undefined`.
+
+### `type`
+
+`'classic'` | `'inline'`
+
+By default, the CKEditor 4 React component creates a {@link guide/dev/framed/README classic editor}. To create an {@link guide/dev/inline/README inline editor}, add the `type` property with the value of `inline` to the `<CKEditor />` tag:
+
+```jsx
+<CKEditor
+	data="<p>Some initial data</p>"
+	type="inline"
+/>
+```
+
+You can also explicitly set the `type` property to `classic` to create the classic editor:
+
+```jsx
+<CKEditor
+	data="<p>Some initial data</p>"
+	type="classic"
+/>
+```
+
+Every other value of the `type` property will be treated as `classic`.
+
+For more details, look at {@link guide/dev/ckeditor_js_load/README Loading CKEditor Script article}.
+
+Defaults to `'classic'`.
+
+### `onBeforeLoad`
+
+`Function`
+
+Callback function with single argument: `CKEDITOR` namespace. It is invoked **each time new editor instance is loaded**, but `CKEDITOR` object always refers to the same namespace.
+
+```jsx
+<CKEditor
+	name="editorOne"
+	data="<p>Editor's content</p>"
+	onBeforeLoad={ CKEDITOR => {
+			// Add external `placeholder` plugin which will be available for each
+			// editor instance on the page.
+			CKEDITOR.plugins.addExternal( 'placeholder', '/path/to/the/placeholder/plugin', 'plugin.js' );
+		}
+	}
+/>
+
+<CKEditor
+	name="editorTwo"
+	data="<p>Editor's content</p>"
+	onBeforeLoad={ CKEDITOR => {
+			// Namespace was already loaded, but this callback will be called after all.
+			// Unnecessary adding same plugin to namespace.
+			CKEDITOR.plugins.addExternal( 'placeholder', '/path/to/the/placeholder/plugin', 'plugin.js' );
+		}
+	}
+/>
+```
+The above example tries to load the same plugin two times to the same namespace.
```

Comment:
It's more readable when the code description is given above the snippet.

Reply:
Moved description above the code snippet.